### ISSUE TITLE
[corpus-tooling] explain pass loses inner reason for wrapper-only break_reasons (37 occurrences)

### ISSUE THREAD

**Issue by penguinwu** [penguinwu/oss-model-graph-break-corpus#123]
## Summary

Our explain pass loses the underlying root cause for some break_reasons. We see entries that contain ONLY the wrapper text ("Failed to handle graph break gracefully" or "Cannot resume from graph break") with no inner Explanation / Operator detail beneath. These end up as `failed_graceful_only` (32 occurrences across 13 models) and `cannot_resume_only` (5 occurrences across 3 models) — un-attributable to any specific Dynamo limitation.

## Why this matters

Without the inner reason, we can't:
1. Re-attribute the break to a specific tracked issue
2. File a per-root-cause sub-issue (the umbrella-split policy needs the underlying reason to apply)
3. Tell whether the break is a known pattern OR something new

## Suspected root cause

`sweep/worker.py` calls `torch._dynamo.explain()` and serializes the resulting `BreakReason` objects into `explain_checkpoint.jsonl`. For some breaks, the inner Explanation field is missing in the serialized output even though Dynamo internally has it. Likely causes:

- Field-loss during the serialization round-trip (some attrs not preserved)
- Truncation of long reason strings
- Dynamo itself surfacing only the wrapper for certain code paths (less likely)

## Investigation steps

1. Find one model where `failed_graceful_only` appears (e.g., from the affected list in #102's split) and run `worker.py` on it directly with `TORCHDYNAMO_VERBOSE=1` and `TORCH_LOGS="+dynamo"`
2. Capture the FULL break_reason output Dynamo produces (not what we serialize) and diff
3. Patch the serializer in `sweep/explain.py` (or wherever `_serialize_break_reason` lives) to preserve the missing fields

## Affected models (sample — from #102 split)

To pull a fresh list:
```bash
python3 tools/sweep_compare.py --pattern "Failed to handle graph break gracefully" \
  --baseline sweep_results/nightly/2026-04-26 \
  --current  sweep_results/nightly/2026-05-03
```
Then look at break_reasons in `explain_checkpoint.jsonl` that contain ONLY the wrapper text.

## Closure criterion

When `failed_graceful_only` and `cannot_resume_only` buckets in the wrapper-umbrella analysis drop to ZERO occurrences (every wrapper break has an attributable inner reason), close this issue.

Surfaced 2026-05-04 during umbrella split of #102 + #103.

**Comment by penguinwu** [comment]
## Closing — diagnosis was wrong; not an explain-pass bug

When I filed this issue 2026-05-04, I claimed `failed_graceful_only` (32 occurrences) and `cannot_resume_only` (5 occurrences) were entries where the explain pass dropped the inner break-reason underneath the wrapper.

**That's wrong.** Verified by sampling the actual `break_reasons` text in `explain_checkpoint.jsonl`: every "wrapper-only" entry actually contains a real inner reason. My own classifier in the umbrella-split bucketing code didn't have rules for these reason patterns, so they fell through to the catch-all `failed_graceful_only` / `cannot_resume_only` buckets — making it look like the inner reason was missing.

## Where the 37 entries actually belong

| Count | Root cause | Tracker |
|---|---|---|
| 14 | `torch.* op returned non-Tensor` | #28 (existing) |
| 7 | Dynamo `missing tp_iter` (iter protocol on Variable that doesn't define it) | not yet tracked — small (7 breaks); fold into next umbrella-split if it grows |
| 4 | `torch.nn.Parameter()` constructor inside compile | #27 (existing) |
| 4 | Uninitialized `nn.Module` | also #27 territory |
| 4 | Unsupported context manager (NOT `lock`) | adjacent to #11 — different ctx type, may need separate tracker if it stays at this size |
| 2 | `Attempted to wrap RNN, GRU, or LSTM` | not yet tracked — small |
| 1 | `requires_grad_()` intermediate leaked as output | #19 |
| 1 | `torch.Generator` method | #26 |

## Lesson

When I see "wrapper-only" entries, the FIRST check should be "did my classifier miss the inner reason?" — not "is the explain pass dropping data?" The serializer is fine. The classifier is incomplete.

Adding to the weekly-sweep-brief methodology: when classifying a wrapper-bucket as "no inner reason captured," sample the actual break_reason text first to confirm the inner reason is genuinely missing vs unclassified.

Closing as not_planned (wrong diagnosis).
